From 4557d033501fc6a9de2a87ff4e436c92631370cc Mon Sep 17 00:00:00 2001
From: Haikun Wang <Haikun.Wang@freescale.com>
Date: Mon, 11 May 2015 19:38:33 +0800
Subject: [PATCH 053/452] mtd: spi-nor: Fix compile issue of JEDEC_MFR

Macro JEDEC_MFR has differnet argument between 3.16 and 3.19, we didn't change
the arguments when port DSPI/QSPI related patches from 3.16 to 3.19.

Signed-off-by: Haikun Wang <haikun.wang@freescale.com>
Change-Id: Ic14a4f5f5098e36d965d7bf1ebb99b8aa5279571
Reviewed-on: http://git.am.freescale.net:8181/36166
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Stuart Yoder <stuart.yoder@freescale.com>
(cherry picked from commit 494f3ce3552d505cf19ffcceec8bce8bf439d14b)
[Stuart: resolved merge conflict]
Signed-off-by: Stuart Yoder <stuart.yoder@freescale.com>
[Xulin: Original patch taken from FSL LS2085 SDK EAR6.0,
LS2085A-SDK-SOURCE-20160304-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/mtd/spi-nor/spi-nor.c |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index 92f2c69..b61b6fa 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -965,11 +965,11 @@ static int micron_quad_enable(struct spi_nor *nor)
 	return 0;
 }
 
-static int set_ddr_quad_mode(struct spi_nor *nor, u32 jedec_id)
+static int set_ddr_quad_mode(struct spi_nor *nor,  struct flash_info *info)
 {
 	int status;
 
-	switch (JEDEC_MFR(jedec_id)) {
+	switch (JEDEC_MFR(info)) {
 	case CFI_MFR_AMD: /* Spansion, actually */
 		status = spansion_quad_enable(nor);
 		if (status) {
@@ -1157,7 +1157,7 @@ int spi_nor_scan(struct spi_nor *nor, const char *name, enum read_mode mode)
 
 	/* DDR Quad/Quad/Dual-read mode takes precedence over fast/normal */
 	if (mode == SPI_NOR_DDR_QUAD && info->flags & SPI_NOR_DDR_QUAD_READ) {
-		ret = set_ddr_quad_mode(nor, info->jedec_id);
+		ret = set_ddr_quad_mode(nor, info);
 		if (ret) {
 			dev_err(dev, "DDR quad mode not supported\n");
 			return ret;
@@ -1177,9 +1177,9 @@ int spi_nor_scan(struct spi_nor *nor, const char *name, enum read_mode mode)
 	/* Default commands */
 	switch (nor->flash_read) {
 	case SPI_NOR_DDR_QUAD:
-		if (JEDEC_MFR(info->jedec_id) == CFI_MFR_AMD) { /* Spansion */
+		if (JEDEC_MFR(info) == CFI_MFR_AMD) { /* Spansion */
 			nor->read_opcode = SPINOR_OP_READ_1_4_4_D;
-		} else if (JEDEC_MFR(info->jedec_id) == CFI_MFR_ST) {
+		} else if (JEDEC_MFR(info) == CFI_MFR_ST) {
 			nor->read_opcode = SPINOR_OP_READ_1_1_4_D;
 		} else {
 			dev_err(dev, "DDR Quad Read is not supported.\n");
-- 
1.7.5.4

